geode-core/src/main/java/org/apache/geode/metrics/internal/InternalDistributedSystemMetricsService.java

```diff
@@ -0,0 +1,275 @@
+    this.publishingServiceLoader = publishingServiceLoader;
+    this.binder = binder;
+    this.persistentMeterRegistries.addAll(persistentMeterRegistries);
+    meterRegistry.config().commonTags(commonTags(system, isClient));
+  }
+
+  /**
+   * Starts this metrics service. The service remains active until {@code stop()} is called.
+   * <p>
+   * This method configures the service's meter registry with:
+   * <ul>
+   * <li>a set of standard process, system, and JVM meters</li>
+   * </ul>
+   *
+   * This method also loads and starts each {@link MetricsPublishingService} found on the classpath.
+   */
+  @Override
+  public void start() {
+    persistentMeterRegistries.forEach(meterRegistry::add);
+
+    binder.bindTo(meterRegistry);
+
+    publishingServices.addAll(publishingServiceLoader.loadServices(MetricsPublishingService.class));
+    publishingServices.forEach(this::startMetricsPublishingService);
+  }
+
+  /**
+   * Returns this metrics service's meter registry.
+   *
+   * @return this metrics service's meter registry
+   */
+  @Override
+  public MeterRegistry getMeterRegistry() {
+    return meterRegistry;
+  }
+
+  @Override
+  public MetricsService.Builder getRebuilder() {
+    return builder;
+  }
+
+  /**
+   * Stops all publishing services and closes this metrics service's meter registry. This method
+   * removes from the metrics service all client meter registries and all registries added by
+   * publishing services. This method does not close any client meter registry or any registry
+   * added to this metrics service by publishing services.
+   * <p>
+   * After this method returns, this metrics service is no longer active, and its meter registry is
+   * closed.
+   */
+  @Override
+  public void stop() {
+    closeMeterBinder();
+    clearAndCloseMeterRegistry();
+  }
+
+  @Override
+  public void addSubregistry(MeterRegistry subregistry) {
+    meterRegistry.add(subregistry);
+  }
+
+  @Override
+  public void removeSubregistry(MeterRegistry subregistry) {
+    meterRegistry.remove(subregistry);
+  }
+
+  private static Set<Tag> commonTags(InternalDistributedSystem system, boolean isClient) {
+    Set<Tag> commonTags = new HashSet<>();
+
+    if (!isClient) {
+      String systemId = String.valueOf(system.getConfig().getDistributedSystemId());
+      commonTags.add(Tag.of("cluster", systemId));
+    }
+
+    String memberName = system.getName();
+    requireNonNull(memberName);
+    if (!memberName.isEmpty()) {
+      commonTags.add(Tag.of("member", memberName));
+    }
+
+    String hostName = system.getDistributedMember().getHost();
+    requireNonNull(hostName);
+
+    if (hostName.isEmpty()) {
+      throw new IllegalArgumentException("Host name must not be empty");
+    }
+    commonTags.add(Tag.of("host", hostName));
+    return commonTags;
+  }
+
+  private void startMetricsPublishingService(MetricsPublishingService service) {
+    try {
+      service.start(this);
+    } catch (Exception thrown) {
```

Comment:
I think the Exception handling in these classes is fine. As long as we're not catching Error or Throwable then we can avoid direct usage of SystemFailure like you see in other code.